In [1]:
import pandas as pd
import numpy as np

In [3]:
df_train_data = pd.read_csv('train_data.csv')
df_test_data = pd.read_csv('test_data.csv')
df_train_labels = pd.read_csv('train_labels.csv')
df_fitness_values = pd.read_csv('fitness_values.csv')

In [4]:
df_train_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14446880 entries, 0 to 14446879
Data columns (total 6 columns):
patient_id         object
event_name         object
event_time         int64
specialty          object
plan_type          object
patient_payment    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 661.3+ MB


In [5]:
df_test_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6256395 entries, 0 to 6256394
Data columns (total 6 columns):
patient_id         object
event_name         object
event_time         int64
specialty          object
plan_type          object
patient_payment    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 286.4+ MB


In [6]:
df_train_labels.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16683 entries, 0 to 16682
Data columns (total 2 columns):
patient_id      16683 non-null object
outcome_flag    16683 non-null int64
dtypes: int64(1), object(1)
memory usage: 260.8+ KB


In [7]:
df_fitness_values.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41525 entries, 0 to 41524
Data columns (total 6 columns):
feature_name               41525 non-null object
avg_1                      41525 non-null float64
avg_0                      41525 non-null float64
sd_1                       41525 non-null float64
sd_0                       41525 non-null float64
coefficient_of_variance    41525 non-null float64
dtypes: float64(5), object(1)
memory usage: 1.9+ MB


In [16]:
df_train_data['patient_id'].describe()

count        14446880
unique          16683
top       patient_299
freq            25526
Name: patient_id, dtype: object

We have observed that no. of unique patient_id in df_train_data is 16683 which is equivalent to rows in df_train_labels

In [18]:
df_train = df_train_data.merge(df_train_labels, on='patient_id',how='inner')

In [20]:
df_train.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 14446880 entries, 0 to 14446879
Data columns (total 7 columns):
patient_id         object
event_name         object
event_time         int64
specialty          object
plan_type          object
patient_payment    float64
outcome_flag       int64
dtypes: float64(1), int64(2), object(4)
memory usage: 881.8+ MB


In [28]:
df_train.head(6)

patient_id event_name  event_time specialty plan_type  patient_payment  \
0  patient_1    event_1          47    spec_1    plan_1              0.0   
1  patient_1    event_1          50    spec_1    plan_1              0.0   
2  patient_1    event_1          47    spec_1    plan_1              0.0   
3  patient_1   event_37         235    spec_1    plan_1              0.0   
4  patient_1    event_1          38    spec_1    plan_1              0.0   
5  patient_1   event_43         107    spec_1    plan_1              0.0   

   outcome_flag  
0             0  
1             0  
2             0  
3             0  
4             0  
5             0

<img src= 'ZS_Fitness_Value.png'>

In [25]:
df_fitness_values.head(10)

feature_name       avg_1       avg_0        sd_1  \
0   recency__event_name__event_1  415.562260  402.517278  315.016268   
1   recency__event_name__event_2  334.024691  308.916367  312.396980   
2   recency__event_name__event_3  430.727273  224.280543  277.010141   
3   recency__event_name__event_4  334.842105  326.087912  223.508206   
4   recency__event_name__event_5  478.988636  455.197872  306.338123   
5   recency__event_name__event_6  475.673469  517.639535  313.416881   
6   recency__event_name__event_7  401.150515  393.824806  308.351280   
7   recency__event_name__event_8  480.220472  477.448143  310.856429   
8   recency__event_name__event_9  514.666667  247.758064  425.351463   
9  recency__event_name__event_10  414.602778  435.794798  313.626875   

         sd_0  coefficient_of_variance  
0  307.536555                 1.007895  
1  298.470307                 1.033075  
2  274.041707                 1.899904  
3  305.235029                 1.402317  
4  315.533077                 1.083849  
5  300.972687                 0.882442  
6  305.504411                 1.009197  
7  318.637462                 1.030983  
8  302.568333                 1.477658  
9  324.751066                 0.985116